syntax = "proto3";

message Person {
  int32 person_id = 1;
  string first_name = 2;
  string last_name = 3;
  string company_name = 4;
}

message Empty {
}

message PersonId {
  int32 person_id = 1;
}

message PersonList {
  repeated Person persons = 1;
}

service PersonService {
    rpc Create(Person) returns (Person) {}
    rpc Retrieve(PersonId) returns (Person) {}
    rpc RetrieveAll(Empty) returns (PersonList) {}
}